coverage/PENDING_SUBMIT/amdvlk/0521-COVERAGE-tiny-ptr-vector-h-85/cov-nested-loops-float-bits-to-int-increment-array.amber

```diff
@@ -0,0 +1,441 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific LLVM code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+#
+# #define _int_5 _GLF_uniform_int_values[0]
+# #define _int_3 _GLF_uniform_int_values[1]
+# #define _int_4 _GLF_uniform_int_values[2]
+# #define _int_6 _GLF_uniform_int_values[3]
+# #define _int_7 _GLF_uniform_int_values[4]
+# #define _int_8 _GLF_uniform_int_values[5]
+# #define _int_1 _GLF_uniform_int_values[6]
+# #define _int_0 _GLF_uniform_int_values[7]
+# #define _int_2 _GLF_uniform_int_values[8]
+# #define _int_9 _GLF_uniform_int_values[9]
+# #define _int_1065353216 _GLF_uniform_int_values[10]
+#
+# precision highp int;
+# precision highp float;
+#
+# // Contents of _GLF_uniform_int_values: [5, 3, 4, 6, 7, 8, 1, 0, 2, 9, 1065353216]
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     int _GLF_uniform_int_values[11];
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     // a becomes 1065353216.
+#     int a = floatBitsToInt(1.0);
+#     int arr[9] = int[9](_int_1, _int_2, _int_3, _int_4, _int_5, _int_6, _int_7, _int_8, _int_9);
+#
+#     // During the loop only arr[8] changes to zero.
+#     for(int i = 0; i < 5; i ++)
+#     {
+#         for(int j = 0; j < 5; j ++)
+#         {
+#             int idx = clamp(a, 0, 8);
+#             arr[idx] = _int_0;
+#
+#             for(int k = _int_0; k < i; k ++)
+#             {
+#                 for(int l = 0; l < j; l ++)
+#                 {
+#                     switch(a - _int_1065353216)
```

Comment:
Add a comment to spell out the fact that a - _int_blah will be zero, which is not one of the switch cases, and thus the switch statement doesn't do anything.

Reply:
a gets incremented later and all the cases do get executed.

Reply:
I see - can you add some more commentary to explain what happens to arr[idx]? I think the point is that idx is guaranteed to be 8, so only arr[8] changes. But the current comment says that only arr[8] changes to zero, which made me think that the switch cases didn't have any effect. Can you make it more explicit when arr[8] changes and what value it ends up having?